### ISSUE TITLE
Helm Private Repository is not working after upgrade to fleet 0.4.0

### ISSUE THREAD

**Issue by matteogazzadi** [rancher/fleet#1044]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have multiple GitRepos where every fleet.yaml is installing an Helm Chart from a private Helm Repository (running on Chart Museum) protected via HTTP Basic Auth.

Until Rancher v2.6.8 everything was working fine. After upgrading to Rancher v2.6.9, all GitRepo went in error state mentioning that fetch from Helm Repo is not possible due to the following error:

`Time="2022-10-21T11:35:34Z" level=info msg="updated: fleet-default/password-file" time="2022-10-21T11:35:34Z" level=fatal msg="failed to read helm repo from https://xxxxxx/index.yaml, error code: 401, response body: {\"error\":\"unauthorized\"}"`

I have another Rancher instance running on v2.6.8 that is working fine and still able to connect to the private Helm Repository.

I have checked and the http auth secret configured in both environment is still the same.

Now we have to rollback to v2.6.8 due to this.

### Expected Behavior

Being able to install helm chart via a private repository protected by HTTP Basic authentication.

### Steps To Reproduce

1. Setup a private Helm Repository protected via HTTP Basic Authentication
2. Create on your rancher cluster a Secret contain the basic auth credentials in fleet-default namespace
3. Add a GitRepo containing at least one fleet.yaml file pulling helm chart from the private repository

### Environment

```markdown
- Architecture:  AKS 1.23.6
- Fleet Version: 0.4.0
- Cluster:
  - Provider: Azure
  - Options:
  - Kubernetes Version: 1.23.6
```

**Comment by mdekoster** [comment]
Also the caBundle is not working in 0.4.0. We have downgraded to Fleet 0.3.11 to get Fleet running again.

On the Rancher Management Cluster go to the Apps - Installed Apps menu and select Fleet 0.3.11.

**Comment by ewokpoacher** [comment]
Also having the same problem, only fleet doesn't show up in my Rancher Apps for downgrading.

**Comment by papanito** [comment]
I have the same problem here, but I also get this message for most of the gitrepos:

```
Get \"https://helm.intra/virtual-helm-bitnami/index.yaml\": x509: certificate signed by unknown authority
```

**Comment by manno** [comment]
This is a bug in 0.4.0, it should be fixed in newer version.

**Comment by papanito** [comment]
For which Rancher version fleet 0.5.0 will be available?

**Comment by thardeck** [comment]
> For which Rancher version fleet 0.5.0 will be available?

Fleet 0.5.0 will be part of Rancher 2.7.0.

**Comment by thardeck** [comment]
We have released Fleet 0.4.1 with two bug fixes, including the fix for this issue: https://github.com/rancher/fleet/releases/tag/v0.4.1
